File: Valera and Antique Items - 441A/main.cpp
#include <bits/stdc++.h>
#define ll long long int
using namespace std;

void Fast()
{
    cin.tie(0);
    cin.sync_with_stdio(0);
}

int main()
{
    Fast();
    int n, current; cin >> n >> current;
    vector<int> total;
    for (int i = 1; i <= n; ++i)
    {
        int seller; cin >> seller;
        bool flag = false;
        for (int j = 0; j < seller; ++j)
        {
            int temp; cin >> temp;
            if (current > temp)
                flag = true;
        }
        if (flag)
            total.emplace_back(i);
    }
    cout << total.size() << "\n";
    for (auto &it : total)
        cout << it << " ";
    return 0;
}

File: Valera and Antique Items - 441A/main.py
n, current = map(int, input().split())
ans = []
for i in range(1, n+1, 1):
    array = list(map(int, input().split()))
    array = array[1:]
    flag = False
    for j in array:
        if j < current:
            flag = True
            break
    if flag:
        ans.append(i)

print(len(ans))
for i in ans:
    print(i, "", end = "")
